crates/cxx-qt-gen/src/generator/rust/qobject.rs

```diff
@@ -116,6 +124,15 @@ fn generate_qobject_definitions(qobject_idents: &QObjectNames) -> Result<Generat
         }
     };
 
+    let base_upcast = if let Some(base) = base {
+        let base_name = type_names
+            .lookup(&format_ident!("{base}"))?
+            .rust_unqualified();
+        quote! { impl cxx_qt::Upcast<#base_name> for #rust_struct_name_rust {} }
```

Comment:
does the compiler or clippy complain about the type not being used? i think Leon needed to add a allow unused to the trait impl or trait itself or something?

Reply:
Not sure, He just said that using that trait prevents dead code warnings for the base type declaration which is now mandatory

Reply:
I think if the trait is coming from another crate, it probably can't figure out that this is unused, so shouldn't complain. :crossed_fingers: 
Otherwise the `#[allow(dead_code)]` needs to be on the trait in the cxx_qt crate itself.